src/graph/util/ExpressionUtils.cpp

```diff
@@ -415,30 +417,67 @@ Expression *ExpressionUtils::reduceUnaryNotExpr(const Expression *expr) {
 
 Expression *ExpressionUtils::rewriteRelExpr(const Expression *expr) {
   ObjectPool *pool = expr->getObjPool();
-  // Match relational expressions containing at least one arithmetic expr
-  auto matcher = [](const Expression *e) -> bool {
-    if (e->isRelExpr()) {
-      auto relExpr = static_cast<const RelationalExpression *>(e);
-      if (isEvaluableExpr(relExpr->right())) {
-        return true;
+  QueryExpressionContext ctx(nullptr);
+
+  auto checkArithmExpr = [&ctx](const ArithmeticExpression *arithmExpr) -> bool {
+    auto lExpr = const_cast<Expression *>(arithmExpr->left());
+    auto rExpr = const_cast<Expression *>(arithmExpr->right());
```

Comment:
For constant expression, you could call its method `value()` directly, so that you don't need to call const_cast.

Reply:
Here we are still not sure about the type of `arithmExpr->left()` and `arithmExpr->right()`